xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/webjar/xwiki-image/imageWizard.js

```diff
@@ -48,6 +48,11 @@ define('imageWizard', ['imageSelector', 'imageEditor'], function(imageSelector,
     if (CKEDITOR.currentInstance) {
       params.currentDocument = CKEDITOR.currentInstance.config.sourceDocument.documentReference;
     }
+
+    if (params.setImageData) {
+      return $.Deferred().resolve(params.setImageData);
```

Comment:
Is it mandatory to use jquery deferred here? Javascript no provides a promise api (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise)

Reply:
I believe it was until the changes Marius made to the CKEditor integration to allow for newest javascript features. I implemented that part before he did the change. jquery deferred is used in other ckeditor plugins we defined. We could replace it indeed. I'll change it here.